R/lockbox.R

```diff
@@ -70,13 +52,30 @@ lockbox.list <- function(lock, env) {
   })
 }
 
-#' @export
-lockbox.default <- function(obj) {
-  stop(
-    "Invalid parameters passed to ", sQuote("lockbox"), " method: ",
-    "must be a ", sQuote("character"), " or ", sQuote("list"), " but ",
-    "instead I got a ", sQuote(class(obj)[1]), "."
-  )
+
+parse_lock <- function(lock, env = getOption("lockbox.env", "!packages")) {
+  if (is.character(lock) && length(lock) == 1) {
+    handle_float_version <- function(x) {
+      if (is.numeric(x)) { sprintf("%.1f", x) }
+      else { x }}
+    lock <- yaml::yaml.load_file(lock, handlers = list("float#fix" = handle_float_version))
+  }
+  else if (!is.list(lock)) {
+  stop("Invalid parameters passed to ", sQuote("lockbox"), " method: ",
+       "must be a ", sQuote("character"), " or ", sQuote("list"), " but ",
+       "instead I got a ", sQuote(class(lock)[1]), ".")
+  }
+  if (is.null(lock$packages)) stop("Invalid config. Make sure your config format is correct")
+  lock <- if (identical(env, "!packages") || is.null(lock[[env]])) {
```

Comment:
The style guide preference here is to assign `lock` in the body for readability.